src/pew/pygobject_gtk/__init__.py

```diff
@@ -0,0 +1,65 @@
+import logging
+
+import gi
+gi.require_version('Gio', '2.0')
+gi.require_version('Gtk', '3.0')
+from gi.repository import Gio, Gtk
+
+
+def run_on_main_thread(func, *args, **kwargs):
+    return func(*args, **kwargs)
```

Comment:
I don’t know any of the context for this, but from a Python/GLib point of view, I don’t think this will run something on the main thread — I think it’ll just run `func` on the current thread.

To call a function on the main thread using the GLib main loop, use `GLib.MainContext.invoke(None, func, data)` (you might need to marshal `*args` and `**kwargs` into `data` and then call `func` from a wrapper function which demarshals them again). The wrapper function should return `False` (see [the documentation](https://developer.gnome.org/glib/stable/glib-The-Main-Event-Loop.html#g-main-context-invoke)).

Reply:
Thanks! Definitely seems it should be doing that :)